### PR TITLE
Implemented conditions for account controller

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR creates the scaffolding for using conditions in the account controller.

### CODE REVIEW

File: pkg/apis/aws/v1alpha1/account_types.go [hunk 1]
```diff
@@ -69,6 +70,13 @@ type AccountCondition struct {
 // AccountConditionType is a valid value for AccountCondition.Type
 type AccountConditionType string
 
+const (
```
Comment (jharrington22) [inline pkg/apis/aws/v1alpha1/account_types.go]: @rafael-azevedo pending too?

File: pkg/controller/account/account_controller.go [hunk 2]
```diff
@@ -41,6 +41,9 @@ const (
 	awsSecretName           = "aws-config"
 	awsAMI                  = "ami-000db10762d0c4c05"
 	awsInstanceType         = "t2.micro"
+	AccountCreating         = "Creating"
```
Comment (jharrington22) [inline pkg/controller/account/account_controller.go]: I think we'll need comments for all of these to keep linting happy?

File: pkg/apis/aws/v1alpha1/account_types.go (hunk 1 above)
Comment (rafael-azevedo) [reply pkg/apis/aws/v1alpha1/account_types.go]: added

File: pkg/controller/account/account_controller.go (hunk 2 above)
Comment (rafael-azevedo) [reply pkg/controller/account/account_controller.go]: added comments

**Pull request merged**